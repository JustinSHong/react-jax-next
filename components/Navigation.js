import React from 'react';
import Link from 'next/link';
const Navigation = (props) => {
    return (
        <nav>
            <Link prefetch href='/'><a>Home</a></Link>
            <Link prefetch href='/characters'><a>Characters</a></Link>
            <Link prefetch href='/planets'><a>Planets</a></Link>
            <style jsx>{`
                nav {
                    display: flex;
                    background-color: black;
                    justify-content: space-around;
                    flex-wrap: wrap;
                }
                a {
                    color: white;
                    font-size: 3rem;
                }
                a:hover {
                    color: grey;
                    transition: 0.3s;
                }
            `}</style>
        </nav>
    );
};

export default Navigation;