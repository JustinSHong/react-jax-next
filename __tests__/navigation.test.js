import React from "react";
import { shallow } from "enzyme";
import { create } from "react-test-renderer";
import Router from "next/router";
// Components
import Navigation from "../components/Navigation";

const mockedRouter = { push: () => {}, prefetch: () => {} };
Router.router = mockedRouter;

describe("with snapshot testing", () => {
    it("renders correctly", () => {
        const component = create(<Navigation />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
