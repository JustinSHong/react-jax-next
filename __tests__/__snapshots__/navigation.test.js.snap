// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with snapshot testing renders correctly 1`] = `
<nav
  className="jsx-79800652"
>
  <a
    className="jsx-79800652"
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
  <a
    className="jsx-79800652"
    href="/characters"
    onClick={[Function]}
  >
    Characters
  </a>
  <a
    className="jsx-79800652"
    href="/planets"
    onClick={[Function]}
  >
    Planets
  </a>
</nav>
`;
